dirsrvtests/tests/suites/acl/aci_excl_filter_test.py

```diff
@@ -0,0 +1,150 @@
+from lib389.idm.domain import Domain
+from lib389.idm.account import Accounts
+
+pytestmark = pytest.mark.tier1
+
+logging.getLogger(__name__).setLevel(logging.DEBUG)
+log = logging.getLogger(__name__)
+
+@pytest.fixture(scope="function")
+def add_anon_aci_access(topo, request):
+    # Add anonymous access aci
+    ACI_TARGET = "(targetattr != \"userpassword\")(target = \"ldap:///%s\")" % (DEFAULT_SUFFIX)
+    ACI_ALLOW = "(version 3.0; acl \"Anonymous Read access\"; allow (read,search,compare)"
+    ACI_SUBJECT = "(userdn=\"ldap:///anyone\");)"
+    ANON_ACI = ACI_TARGET + ACI_ALLOW + ACI_SUBJECT
+    suffix = Domain(topo.standalone, DEFAULT_SUFFIX)
+
+    try:
+        suffix.add('aci', ANON_ACI)
+    except ldap.TYPE_OR_VALUE_EXISTS:
+        pass
+    def fin():
+        suffix.delete()
+    request.addfinalizer(fin)
+
+
+def add_ou_entry(topo, name, myparent):
+
+    ou_dn = 'ou={},{}'.format(name, myparent)
+    ou = OrganizationalUnit(topo.standalone, dn=ou_dn)
+    assert ou.create(properties={'ou': name})
+    log.info('Organisation {} created for ou :{} .'.format(name, ou_dn))
+
+
+def add_user_entry(topo, user, name, pw, myparent):
+
+    dn = 'ou=%s,%s' % (name, myparent)
+    properties = {
+        'uid': name,
+        'cn': 'admin',
+        'sn': name,
+        'uidNumber': '1000',
+        'gidNumber': '2000',
+        'homeDirectory': '/home/{}'.format(name),
+        'telephonenumber': '+1 222 333-4444',
+        'userpassword': pw,
+        }
+
+    assert user.create(properties=properties)
+    log.info('User created for dn :{} .'.format(dn))
+    return user
+
+
+def test_aci_with_exclude_filter(topo, add_anon_aci_access):
+    """
+       Test an ACI(Access control instruction) which contains an extensible filter.
+    :id: 238da674-81d9-11eb-a965-98fa9ba19b65
+    :setup: Standalone instance
+    :steps:
+        1. Bind to a new Standalone instance
+        2. Generate text for the Access Control Instruction(ACI) and add to the standalone instance
+           -Create a test user 'admin' with a marker -> deniedattr = 'telephonenumber'
+        3. Create 2 top Organizational units (ou) under the same root suffix
+        4. Create 2 test users for each Organizational unit (ou) above with the same username 'admin'
+        5. Bind to the Standalone instance as the user 'admin' from the ou created in step 4 above
+           - Search for user(s) ' admin in the subtree that satisfy this criteria:
+               DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, cn_filter, [deniedattr, 'dn']
+        6.  The search should return 2 entries with the username 'admin'
+        7.  Verify that the users found do not have the --> deniedattr = 'telephonenumber' marker
+    :expectedresults:
+        1. Operation should be successful
+        2. Operation should be successful
+        3. Operation should be successful
+        4. PASS - users found do not have the --> deniedattr = 'telephonenumber' marker
```

Comment:
You have only 4 'expectedresults' but you have 7 'steps'.

I think our test documentation system requires a full alignment (7 steps - 7 expected results)

Reply:
Thank you -fixed the missing results.